### ISSUE TITLE
[Bug]: how to remove --filesystem setted permissions.

### ISSUE THREAD

**Issue by ziwen1002** [flatpak/flatpak#5854]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a bug that matches the one I want to file, without success.
- [X] If this is an issue with a particular app, I have tried filing it in the appropriate issue tracker for the app (e.g. under https://github.com/flathub/) and determined that it is an issue with Flatpak itself.
- [X] This issue is not a report of a security vulnerability (see [here](https://github.com/flatpak/flatpak/blob/main/SECURITY.md) if you need to report a security issue).

### Flatpak version

1.15.8

### What Linux distribution are you using?

Arch Linux

### Linux distribution version

Linux lxx 6.9.5-arch1-1 #1 SMP PREEMPT_DYNAMIC Sun, 16 Jun 2024 19:06:37 +0000 x86_64 GNU/Linux

### What architecture are you using?

x86_64

### How to reproduce

1. run `flatpak override --user --filesystem=host`
2. run `flatpak override --user --filesystem=host --filesystem=host-os org.ardour.Ardour`
3. run `flatpak override --user --nofilesystem=host org.ardour.Ardour`
4. run `flatpak override --user --show org.ardour.Ardour`

### Expected Behavior

`flatpak override --user --show org.ardour.Ardour` output:

```
[Context]
filesystems=host-os;
```

`filesystems=host` inherits from the global.

### Actual Behavior

`flatpak override --user --show org.ardour.Ardour` output:

```
[Context]
filesystems=host-os;!host;
```

The application does not have `filesystems=host` permissions.

I guess -nofilesystem is a restricted permission, but how to remove --filesystem setted permissions.

### Additional Information

I know i can modify the config file `~/.local/share/flatpak/overrides/org.ardour.Ardour` to delete it.

But I don't want to, I need a uniform way to modify permissions.

**Comment by hfiguiere** [comment]
`!host-os` is actually a negation. I don't see what the problem is here.

**Comment by ziwen1002** [comment]
> `!host-os` is actually a negation. I don't see what the problem is here. `!host-os` 实际上是一个否定。我不明白这里有什么问题。

yes,  The problem is that there is no unset command. unset --filesystem and unset --nofilesystem.

**Comment by hfiguiere** [comment]
that's was `--nofilesystem` is for, but it's a bit more subtle. There is also `--nofilesystem=host:reset`.

Without understanding what your actual goal is here, it's hard. Ardour has `home` as a default permission.

**Comment by ziwen1002** [comment]
I agree with what --nofilesystem does.

There should be 3 ways to set up rules:

- set allow
- set deny
- unset allow or unset deny, to inherit global

But now there is no unset operation.

I've just started giving an app access to a  path, and then made that path globally accessible,  I want to remove the app's permissions so that it inherits the global rules, but I can't do that through cli, I have to modify the config file.

**Comment by ziwen1002** [comment]
> that's was `--nofilesystem` is for, but it's a bit more subtle. There is also `--nofilesystem=host:reset`.
> 
> Without understanding what your actual goal is here, it's hard. Ardour has `home` as a default permission.

I agree with what --nofilesystem does.

There should be 3 ways to set up rules:

- set allow
- set deny
- unset allow or unset deny, to inherit global

But now there is no unset operation.

I've just started giving an app access to a path, and then made that path globally accessible, I want to remove the app's permissions so that it inherits the global rules, but I can't do that through cli, I have to modify the config file.

**Comment by smcv** [comment]
> I've just started giving an app access to a path, and then made that path globally accessible, I want to remove the app's permissions so that it inherits the global rules, but I can't do that through cli, I have to modify the config file.

That seems like a reasonable feature request for `flatpak override`.

At the moment, you can force one parameter to a specific value with something like `flatpak override --filesystem=/foo` or `flatpak override --nofilesystem=/foo`, and you can clear out *all* overrides with `flatpak override --reset $APP_ID`. But you can't reset the state of `--[no]filesystem=/foo` without also resetting every other override, for example `--filesystem=/bar` or `--env=baz`.

**Comment by chrisawi** [comment]
Cross-referencing to https://github.com/flatpak/flatpak/issues/6035

If we add a replacement for `flatpak override`, that would be a good opportunity to do this better.
